grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALGeneratedRelation.java

```diff
@@ -43,9 +43,9 @@
         this.factory = new StandardRepresentationFactory();
     }
 
-    Representation getNewGeneratedRelation(String assertionID, Optional<TypeName> relationType) {
+    Representation getNewGeneratedRelation(String firstID, String secondID, String assertionID, Optional<TypeName> relationType) {
```

Comment:
In my opinion these IDs should be `ConceptID`s and you should only be calling `getValue()` in here because this is really the only place you *need* the `String` representation.